functions/Test-DbaDatabseBestPractice

```diff
@@ -0,0 +1,305 @@
+		}
+
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+        $processed = New-Object System.Collections.ArrayList
+
+		foreach ($server in $sqlserver) 
+		{
+            if ($server -notin $processed)
+			{
+				$null = $processed.Add($server)
+				Write-Verbose "Connecting to $server"
+			}
+			else
+			{
+				continue
+			}
+
+			Write-Verbose "Attempting to connect to $sqlserver"
+			
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+			}
+			catch
+			{
+				Write-Warning "Can't connect to $sqlserver or access denied. Skipping."
+				continue
+			}
+			
+            $databases = $server.Databases;
+
+			if ($databases.count -eq 0)
+			{
+				$dbs = $server.Databases | Where-Object { $databases -contains $_.Name }
+			}
+			else
+			{
+				if ($IncludeSystemDbs -eq $true)
+				{
+					$dbs = $server.Databases
+				}
+				else
+				{
+					$dbs = $server.Databases | Where-Object { $systemdbs -notcontains $_.Name }
+				}
+			}
+			
+			foreach ($db in $dbs)
+			{
+				$autogrowthbps = Test-AutoGrowth $server, $db
+				$falsecount = $autogrowthbps | Where-Object { $_.IsBestPractice -eq $false }
+				
+				if ($falsecount.count -gt 0)
+				{
+					$IsBestPracticeAutoGrowth = $false
+				}
+				else
+				{
+					$IsBestPracticeAutoGrowth = $true
+				}
+				
+				$object = New-Object PSObject -Property @{
+					Server = $sqlserver
+					Database = $db.Name
+					IsBestPracticeAutoClose = Test-AutoClose $server $db
+					IsBestPracticeAutoShrink = Test-AutoShrink $server, $db
+					IsBestPracticePageVerify = Test-PageVerify $server, $db
+					IsBestPracticeAutoGrowth = $IsBestPracticeAutoGrowth
+					AutoGrowthDetails = $autogrowthbps
+				}
+				$null = $collection.Add($object)
+			}
+		}
+	}
+	
+	END
+	{
+		if ($Detailed -eq $true)
+		{
+			return ($collection | Sort-Object Server, Database | Select-Object *)
+		}
+		else
+		{
+			$bps = 'IsBestPracticeAutoClose', 'IsBestPracticeAutoShrink', 'IsBestPracticePageVerify', 'IsBestPracticeAutoGrowth'
+			
+			$newcollection = @()
+			foreach ($server in $processed)
+			{
+				$ServerResults = $collection | Where-Object { $_.Server -eq $Server }
+				
+				$DBs = $ServerResults | Select-Object Database -Unique
+				
+				foreach ($dbInner in $DBs)
+				{
+                    $IsBestPractice = $true;
+
+                    $properties = $collection | Where-Object { $_.Database -eq $dbInner.Database };
+                    $bpsproperties = $properties | Select -Property *BestPrac* | gm -MemberType NoteProperty | % {$_.Name}
+					
+                    foreach($prop in $bpsproperties)
+                    {
+                        $localtemp = $properties."$prop";
+                        $IsBestPractice = $localtemp -and $IsBestPractice;
+                    }
+
+					$newcollection += [PSCustomObject]@{
+						Server = $server
+						Database = $dbInner.Database
+						IsBestPractice = $IsBestPractice
+					}
```

Comment:
I would ensure you are being consistent in your code. In places you are using `New-Object PSObject -Property @{}` to build a collection, where in these lines you utilize `[PSCustomerObject]`. The PSCustomeObject is preferred (by me at least).